### PR TITLE
[bitnami/redis] Enable PodDisruptionBudgets

### PR DESCRIPTION
**Pull request by fmulero**
### Description of the change

Enabled PodDisruptionBudget by default and little fixes in PDB configuration to keep it aligned with current templates.

### Benefits

PDB protects our applications from voluntary disruption initiated by cluster administrators.
Keep our charts up to date according to our templates.

### Possible drawbacks

None

### Adition information

* [Pod disruption budgets](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#pod-disruption-budgets)
* [Specifying a Disruption Budget for your Application](https://kubernetes.io/docs/tasks/run-application/configure-pdb/)

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by andresbono — APPROVED** [review]

**Comment by wouterh** [comment]
@fmulero: I agree with @fcuello-fudo's comments that there's something wrong with the conditions for creating the PodDisruptionBudgets. Also, for backwards compatibility, I believe `bitnami/redis/templates/sentinel/pdb.yaml` should also coalesce the old (deprecated) pdb values in the same way as is done in `bitnami/redis/templates/replicas/pdb.yaml`. 

This change broke the PodDisruptionBudgets in our sentinel setup. I have upgraded our development environment to chart v19.5.0 and now I have the following PDBs:

```bash
$ kubectl -n redis get pdb -o jsonpath="{.items[].metadata.name} {.items[].spec.selector} {.items[].status.expectedPods}"
redis-master {"matchLabels":{"app.kubernetes.io/component":"master","app.kubernetes.io/instance":"redis","app.kubernetes.io/name":"redis"}} 0
```
It created a `redis-master` pdb (which doesn't match any pods) instead of creating a `redis-node` pdb.

In our staging environment, where we still run chart v19.3.0:

```bash
kubectl -n redis get pdb -o jsonpath="{.items[].metadata.name} {.items[].spec.selector} {.items[].status.expectedPods}"
redis {"matchLabels":{"app.kubernetes.io/instance":"redis","app.kubernetes.io/name":"redis"}} 3
```

The relevant values:
```yaml
pdb:
  create: true
  minAvailable: 2
replica:
  replicaCount: 3
sentinel:
  enabled: true
```

File: bitnami/redis/templates/sentinel/pdb.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if and (eq .Values.architecture "replication") (not .Values.sentinel.enabled) .Values.replica.pdb.create }}
```
[the remark this answers is not in the document]
Comment (fmulero) [reply bitnami/redis/templates/sentinel/pdb.yaml:6]: Yes, you right, the condition is wrong. Thanks for the heads up.

File: bitnami/redis/templates/master/pdb.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if and .Values.master.pdb.create (gt (int64 .Values.master.count) 0) (or (eq .Values.architecture "replication") (not .Values.sentinel.enabled)) }}
```
[the remark this answers is not in the document]
Comment (fmulero) [reply bitnami/redis/templates/master/pdb.yaml:6]: I applied the same condition as in the master deployment. We can enable PDBs with just only one pod

**Comment by fmulero** [comment]
Hi @fcuello-fudo and @wouterh 

Feel free to review https://github.com/bitnami/charts/pull/27080 and continue discussions there

**Pull request merged**